search-service/src/main/kotlin/com/egm/stellio/search/authorization/SubjectReferentialService.kt

```diff
@@ -188,7 +188,9 @@ class SubjectReferentialService(
                 """.trimIndent()
             )
             .bind("subject_id", (sub as Some).value)
-            .allToMappedList { GlobalRole.forKey(it["global_roles"] as String) }
```

Comment:
instead, what about changing the SQL query with:

```
SELECT unnest(global_roles)
FROM subject_referential
WHERE subject_id = :subject_id
```

and leaving the rest of the code untouched.

also, add one (or more) unit test for this function, this kind of problem should appear before the integration tests.